SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx

```diff
@@ -1271,28 +1384,452 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::AddPointToScene(double x,
+        PointType p1;
+        // Read in the source points set
+        for(unsigned int i = 0; i < polyData_source->GetNumberOfPoints(); i += 10){
+            double p[3];
+            polyData_source->GetPoint(i,p);
+            // This is identical to:
+            // polydata->GetPoints()->GetPoint(i,p);
+            p1[0] = p[0];
+            p1[1] = p[1];
+            p1[2] = p[2];
+            sourceLandMarkContainer->InsertElement(id_s, p1);
+            id_s++;
+        }
+        tps->SetSourceLandmarks(sourceLandMarks);
+
+        std::string upFileName(tempFolder);
+        std::string downFileName(tempFolder);
+        std::string crestFileName(tempFolder);
+        upFileName = upFileName + "/model/up" + std::to_string(stepNum + 1) + ".vtk";
+        downFileName = downFileName + "/model/down" + std::to_string(stepNum + 1) + ".vtk";
+        crestFileName = crestFileName + "/model/crest" + std::to_string(stepNum + 1) + ".vtk";
+
+        // read source srep
+        vtkSmartPointer<vtkPolyDataReader> upSpokeReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        upSpokeReader->SetFileName(upFileName.c_str());
+        upSpokeReader->Update();
+        vtkSmartPointer<vtkPolyData> upSpokes = upSpokeReader->GetOutput();
+
+        vtkSmartPointer<vtkPolyDataReader> downSpokeReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        downSpokeReader->SetFileName(downFileName.c_str());
+        downSpokeReader->Update();
+        vtkSmartPointer<vtkPolyData> downSpokes = downSpokeReader->GetOutput();
+
+        vtkSmartPointer<vtkPolyDataReader> crestSpokeReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        crestSpokeReader->SetFileName(crestFileName.c_str());
+        crestSpokeReader->Update();
+        vtkSmartPointer<vtkPolyData> crestSpokes = crestSpokeReader->GetOutput();
+
+        upFileName = tempFolder + "/model/up" + std::to_string(stepNum);
+        std::string upOutputFile(upFileName.c_str());
+        TransformNOutput(tps, upSpokes, upOutputFile);
+        // display upspokes
+
+        downFileName = tempFolder + "/model/down" + std::to_string(stepNum);
+        std::string downOutputFile(downFileName.c_str());
+        TransformNOutput(tps, downSpokes, downOutputFile);
+
+        crestFileName = tempFolder + "/model/crest" + std::to_string(stepNum);
+        std::string crestOutputFile(crestFileName.c_str());
+        TransformNOutput(tps, crestSpokes, crestOutputFile);
+
+    }
+
 }
 
-int vtkSlicerSkeletalRepresentationInitializerLogic::GenerateSrep(std::string& output)
+int vtkSlicerSkeletalRepresentationInitializerLogic::DisplayResultSrep()
 {
-    std::string tempDir(this->GetApplicationLogic()->GetTemporaryPath());
-    output = tempDir +  "/ellipsoid.m3d";
+    // Hide other nodes.
+    HideNodesByClass("vtkMRMLModelNode");
+    const char *tempFolder = this->GetApplicationLogic()->GetTemporaryPath();
+    std::string upFileName(tempFolder);
```

Comment:
tempFolder could be a const std::string,
and then make the files const as well with one line:
Example:
```cpp
const std::string upFileName(tempFolder + "/model/up1.vtk");
```